src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -167,6 +161,27 @@ public class Utils {
         }
     }
 
+    /**
+     * Marks the containing stage with this name as a synthetic stage, with the appropriate context.
+     *
+     * @param stageName
+     * @param context
+     */
+    @Whitelisted
+    static void markSyntheticStage(String stageName, String context) {
+        CpsThread thread = CpsThread.current()
+        CpsFlowExecution execution = thread.execution
+
+        FlowNode currentNode = execution.currentHeads.find { n ->
+            n?.displayName?.equals(stageName)
+        }
+
+        if (currentNode.getAction(TagsAction.class) == null) {
+            currentNode.actions.add(new TagsAction())
+        }
+        currentNode.getAction(TagsAction.class).addTag(SYNTHETIC_STAGE_TAG, context)
```

Comment:
Is there a way to limit the visibility/exposure of a `Step`? i.e., can I create one that only can be used in this plugin? If not, then hacks are really the only way I can see to get what we need - a `Step` that shows up in the snippet generator, etc that does this would be...flawed.